packages/ui/src/components/TransferModal.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import BN from 'bn.js'
+import { BorderRad, Colors } from '../constants'
+import { useApi } from '../hooks/useApi'
+import { useBalances } from '../hooks/useBalances'
+import { useNumberInput } from '../hooks/useNumberInput'
+import { useKeyring } from '../hooks/useKeyring'
+import { Account } from '../hooks/types'
+import { formatTokenValue } from '../utils/formatters'
+import { Modal, ModalBody, ModalFooter, ModalHeader } from './modal'
+import { ButtonPrimaryMedium, ButtonSecondarySmall } from './buttons/Buttons'
+import { AccountInfo } from './AccountInfo'
+
+interface Props {
+  onClose: () => void
+  from: Account
+  to: Account
+}
+
+export function TransferModal({ from, to, onClose }: Props) {
+  const { api } = useApi()
+  const { keyring } = useKeyring()
+  const balances = useBalances([from, to])
+  const [isSending, setIsSending] = useState(false)
+  const [amount, setAmount] = useNumberInput(0)
+
+  if (!balances.hasBalances)
+    return (
+      <Modal>
+        <ModalBody>Loading balances...</ModalBody>
+      </Modal>
+    )
+
+  const transferableBalance = balances?.map[from.address]?.total
+
+  const isOverBalance = new BN(amount).gt(transferableBalance)
+  const isZero = new BN(amount).lte(new BN(0))
+  const isSendDisabled = isSending || isZero || isOverBalance
+
+  const setHalf = () => setAmount(transferableBalance.div(new BN(2)).toString())
+  const setMax = () => setAmount(transferableBalance.toString())
+  const signAndSend = async () => {
+    setIsSending(true)
+
+    if (!api || !keyring) {
+      return
+    }
+
+    await api.tx.balances
```

Comment:
Consider adding a UI test for this modal. I'm not insisting on it, but could trigger some healthy refactoring (eg. not depending on api directly).

Reply:
That makes sense. I'll add them in the next task.